webui/react/src/components/NotebookModal.tsx

```diff
@@ -239,17 +240,20 @@ const NotebookForm:React.FC<FormProps> = (
   const [ templates, setTemplates ] = useState<Template[]>([]);
   const [ resourcePools, setResourcePools ] = useState<ResourcePool[]>([]);
   const [ resourceInfo, setResourceInfo ] = useState<ResourceInfo>(
-    { hasAux: false, hasCompute: true, showResourceType: true },
+    { hasAux: false, hasCompute: true, maxSlots: 1, showResourceType: true },
   );
 
   const calculateResourceInfo = useCallback((selectedPoolName: string | undefined) => {
     const selectedPool = resourcePools.find(pool => pool.name === selectedPoolName);
     if (!selectedPool) {
-      return { hasAux: false, hasCompute: false, showResourceType: true };
+      return { hasAux: false, hasCompute: false, maxSlots: 0, showResourceType: true };
     }
     const hasAuxCapacity = selectedPool.auxContainerCapacityPerAgent > 0;
     const hasComputeCapacity = selectedPool.slotsAvailable > 0
       || (!!selectedPool.slotsPerAgent && selectedPool.slotsPerAgent > 0);
+    const maxSlots = hasComputeCapacity ?
+      (selectedPool.slotsPerAgent && selectedPool.slotsPerAgent > 0 ?
+        selectedPool.slotsPerAgent : 2**16) : 0;
```

Comment:
nit: I think you could use `undefined` to represent "lack of known max slots" and the input element should accept it and behave niely as well. This works too